telemetry/derive/tests/ui.rs

```diff
@@ -0,0 +1,8 @@
+use trybuild::TestCases;
+
+#[test]
+fn ui() {
+    let test_cases = TestCases::new();
+    test_cases.pass("tests/ui_pass/*.rs");
```

Comment:
since this library is used internally passing tests aren't a hard requirement because any compilation issue would be caught immediately at the point of use, otherwise it would be prudent to have some passing tests.

This is just a comment

Reply:
I have considered adding passing tests. The issue is that they don't test anything really. Once there's some more metrics, and some more logic, it would make sense to add some.